src/v/crypto/include/crypto/crypto.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "bytes/bytes.h"
+#include "crypto/exceptions.h"
+#include "crypto/types.h"
+
+#include <cstddef>
+#include <memory>
+
+namespace crypto {
+///////////////////////////////////////////////////////////////////////////////
+/// Digest operations
+///////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Structure used to hold the context of digest operations
+ */
+struct digest_context {
+    ~digest_context() noexcept;
+    class impl;
+    std::unique_ptr<impl> _impl;
```

Comment:
also this is public but consumers would never be able to call method on it since impl is forward declared. I think making the details transparent would solve this too

Reply:
> I think making the details transparent would solve this too

The details of `impl`?  I was trying to hide those :)